maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/NodeHelper.java

```diff
@@ -32,19 +36,38 @@ public class NodeHelper {
 	
 	private final BuildPluginManager pluginManager;
 
-	private Path tempDir;
-	
-	public NodeHelper(MavenProject mavenProject,
-			MavenSession mavenSession,
-			BuildPluginManager pluginManager) {
+	private final Path tempDir;
+
+	private static final Logger logger = LoggerFactory.getLogger(NodeHelper.class);
+
+	public Path getTempDir() {
+		return tempDir;
+	}
+
+	public NodeHelper(MavenProject mavenProject, MavenSession mavenSession, BuildPluginManager pluginManager) {
 		this.mavenProject = mavenProject;
 		this.mavenSession = mavenSession;
 		this.pluginManager = pluginManager;
 		try {
-			this.tempDir = Files.createTempDirectory("openmrs-sdk-node");
-			this.tempDir.toFile().deleteOnExit();
+			Properties sdkProperties = getSdkProperties();
+			boolean isDataSavingMode = Boolean.parseBoolean(sdkProperties.getProperty("enableDataSaving"));
+			String cacheName = "openmrs-sdk-node";
+			if (isDataSavingMode) {
+				logger.info("OpenMRS SDK is running on data saving mode");
```

Comment:
I would not call this a data saving mode because it may be used for other reasons like performance instead of just saving data.